File: src/main/java/com/fendyk/clients/fetch/FetchActivities.java
package com.fendyk.clients.fetch;

import com.fendyk.DTOs.ActivitiesDTO;
import com.fendyk.DTOs.MinecraftUserDTO;
import com.fendyk.DTOs.updates.UpdateActivitiesDTO;
import com.fendyk.Main;
import com.fendyk.clients.FetchAPI;
import okhttp3.Request;
import okhttp3.RequestBody;

import java.util.UUID;

public class FetchActivities extends FetchAPI<ActivitiesDTO, UpdateActivitiesDTO> {
    public FetchActivities(String url) {
        super(url, ActivitiesDTO.class);
    }
}

File: src/main/java/com/fendyk/clients/fetch/FetchActivities.kt
package com.fendyk.clients.fetch

import com.fendyk.DTOs.ActivitiesDTO
import com.fendyk.DTOs.updates.UpdateActivitiesDTO
import com.fendyk.clients.FetchAPI
import java.util.*
import java.util.concurrent.CompletableFuture

class FetchActivities : FetchAPI<UUID, ActivitiesDTO, UpdateActivitiesDTO>(ActivitiesDTO::class.java) {
    override fun update(key: UUID, dto: UpdateActivitiesDTO): CompletableFuture<ActivitiesDTO?> {
        return fetch("/chunks/$key", RequestMethod.PATCH, dto)
    }

}
